### PR TITLE
[NDM] [snmp] Add integration field for TopologyLinkMetadata

### PR DESCRIPTION
**Pull request by zoedt**
### What does this PR do?
https://datadoghq.atlassian.net/browse/NDINT-249

### Describe how to test/QA your changes
https://datadoghq.atlassian.net/wiki/spaces/II/pages/1385365746/Agent+development#QA
https://datadoghq.atlassian.net/wiki/spaces/AI/pages/2947285412/Using+Custom+Datadog+Agents+in+ddev

1. `ddev env start snmp py3.12-false -a registry.ddbuild.io/ci/datadog-agent/agent:v51363080-4856a0ab-7-arm64`
2. `ddev env config edit snmp py3.12-false`
3. `ddev env reload snmp py3.12-false`
4. `ddev env check snmp py3.12-false > snmp-check.txt`

Then you can change community_string from “public” to a specific profile you’d like to test, for example “cisco-ucs”.  (This should refer to the name of the .snmprec file, without the .snmprec suffix.)

You can use these `.snmprec`: 
- [_generic-lldp.snmprec](https://github.com/DataDog/integrations-core/blob/ac011808f171b85a0fc4dce11247d8652187c58f/snmp/tests/compose/data/_generic-lldp.snmprec#L12)
- [aos-lldp.snmprec](https://github.com/DataDog/integrations-core/blob/ac011808f171b85a0fc4dce11247d8652187c58f/snmp/tests/compose/data/aos-lldp.snmprec#L29)
- [cisco-cdp-lldp.snmprec](https://github.com/DataDog/integrations-core/blob/ac011808f171b85a0fc4dce11247d8652187c58f/snmp/tests/compose/data/cisco-cdp-lldp.snmprec#L11)

Check that for a topology link metadata, it gets submitted with `integration` field!

```
...
      "links": [
        {
          "id": "COMP-LD2CNQ24RV:172.18.0.2:7.1",
          "integration": "snmp",
          "local": {
            "device": {
              "dd_id": "COMP-LD2CNQ24RV:172.18.0.2"
            },
            "interface": {
              "dd_id": "COMP-LD2CNQ24RV:172.18.0.2:7",
              "id": "te1/0/7"
            }
          },
          "remote": {
            "device": {
              "id": "82:8a:8c:2f:f8:36",
              "id_type": "mac_address",
              "ip_address": "10.25.0.19",
              "name": "K05-ITV"
            },
            "interface": {
              "id": "gi9",
              "id_type": "interface_name"
            }
          },
          "source_type": "lldp"
        }
      ],
      "namespace": "COMP-LD2CNQ24RV"
    }
  }
```

### CODE REVIEW

File: pkg/collector/corechecks/snmp/integration_topology_test.go [hunk 1]
```diff
@@ -661,6 +661,7 @@ profiles:
         {
             "id": "profile-metadata:1.2.3.4:101.1",
             "source_type": "lldp",
+            "integration": "snmp",
```
Comment (zoedt) [inline pkg/collector/corechecks/snmp/integration_topology_test.go:664]: not sure if there's any other tests to account for?

File: releasenotes/notes/add-integration-field-to-topology-link-metadata-8a35c0a70ac80b1e.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Emit new field: integration for TopologyLinkMetadata for better metrics collection.
+issues:
+  - |
+    List known issues here, or remove this section.
```
Comment (cswatt) [inline releasenotes/notes/add-integration-field-to-topology-link-metadata-8a35c0a70ac80b1e.yaml:14]:
```suggestion
```

Comment (zoedt) [reply releasenotes/notes/add-integration-field-to-topology-link-metadata-8a35c0a70ac80b1e.yaml]: woopsie ty!!

Comment (zoedt) [reply releasenotes/notes/add-integration-field-to-topology-link-metadata-8a35c0a70ac80b1e.yaml]: also love the ur avatar hat (takoyaki??)

**Review by FlorianVeaux — APPROVED** [review]

**Review by AlexandreYang — APPROVED** [review]

**Review by brett0000FF — APPROVED** [review]
Approved with small edit suggestion for the release note. 👍

**Pull request merged**